wolfcrypt/src/cryptocb.c

```diff
@@ -267,13 +292,47 @@ int wc_CryptoCb_RegisterDevice(int devId, CryptoDevCallbackFunc cb, void* ctx)
     dev->cb = cb;
     dev->ctx = ctx;
 
-    return 0;
+    if(cb) {
+        /* Invoke callback with register command */
+        wc_CryptoInfo info= {
```

Comment:
Can this be const? Is it portable?

Reply:
This can't be const as the register command is expected to update the ctx member of this struct to the updated context value.  Typical use case would be passing NULL ctx into the callback, hardware is opened/allocated, info->ctx is updated within the cb with the new value, then this function updates the dev->ctx with the new value.

Portable?  I guess if the struct order changes, then this initialization will fail on C++.  I'll change to a series of sets.

Reply:
Updated to improve portability for initialization.